explainer-server/app/config/Config.scala

```diff
@@ -1,25 +1,51 @@
 package config
 
+import javax.inject.{Inject, Singleton}
+
 import com.amazonaws.auth.{AWSCredentialsProviderChain, InstanceProfileCredentialsProvider}
 import com.amazonaws.auth.profile.ProfileCredentialsProvider
-import com.typesafe.config.ConfigFactory
+import com.amazonaws.regions.{Region, Regions}
+import com.amazonaws.services.kinesis.AmazonKinesisClient
+import play.api.Configuration
+import services.{AWS, AwsInstanceTags}
+
+@Singleton
+class Config @Inject() (conf: Configuration) extends AwsInstanceTags {
+
 
-object Config {
-  val conf = ConfigFactory.load()
+  val stage = readTag("Stage") getOrElse "DEV"
 
-  val stage = conf.getString("stage")
+  def configValueForStage(value: String) = conf.getString(stage+"."+value)
 
-  val publicConfigForStage = conf.getConfig(stage)
+  val pandaDomain = configValueForStage("pandomain.domain").get
 
-  val pandaDomain = publicConfigForStage.getString("pandomain.domain")
+
+  lazy val region = {
+    val r = conf.getString("aws.region").map(Regions.fromName).getOrElse(Regions.EU_WEST_1)
+    Region.getRegion(r)
+  }
 
   val awsCredentialsprovider = new AWSCredentialsProviderChain(
     new ProfileCredentialsProvider("composer"),
     new InstanceProfileCredentialsProvider
   )
 
+  val publishToKinesis = conf.getBoolean("enable.kinesis.publishing") getOrElse true
+
+  val previewKinesisStreamName = if (publishToKinesis) configValueForStage("kinesis.streamName.preview").get else ""
```

Comment:
Thanks. At the moment I'm doing a .get on all essential config which will give a None.get error if the config isn't provided. It would probably be nicer to throw a useful error message with sys.error though.